cmds/core/sync/sync_unix.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2023 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build !linux && !plan9
+// +build !linux,!plan9
+
+package main
+
+import (
+	"fmt"
+	"syscall"
+
+	"golang.org/x/sys/unix"
+)
+
+func sync(data, filesystem bool, args []string) error {
+	switch {
+	case data:
+		return doSyscall(unix.SYS_FDATASYNC, args)
+	case filesystem:
```

Comment:
do you really want to do it this way? Checking flags in a support function? If you just move
filesystem = flag.Bool("filesystem", false, "commit filesystem caches to disk")
to sync_linux.go, then the flag won't even be available on anything but linux.

Reply:
If we move `filesystem = flag.Bool("filesystem", false, "commit filesystem caches to disk")` to sync_linux it will be not available in sync.go and the only way to know if -d option if to use global pointer *filesystem. In that case only linux version will know about -d flag, but we need to do all pointers set/reset and tests. We use the same approach in ls.

So i kinda don't like both approaches, but if you prefer moving filesystem under _linux and make it like ls, I can do that.